### PR TITLE
[threads] Remove mono_threads_create_thread

### PR DESCRIPTION
**Pull request by luhenry**
It is only used to create managed threads, so replace its uses by mono_thread_create_internal.

### CODE REVIEW

File: mono/metadata/appdomain.c [hunk 1]
```diff
@@ -2479,14 +2479,12 @@ unload_thread_main (void *arg)
...
+	MonoInternalThread *internal;
 	int i;
 
-	/* Have to attach to the runtime so shutdown can wait for this thread */
-	/* Force it to be attached to avoid racing during shutdown. */
-	thread = mono_thread_attach_full (mono_get_root_domain (), TRUE);
+	internal = mono_thread_internal_current ();
 
-	mono_thread_set_name_internal (thread->internal_thread, mono_string_new (mono_get_root_domain (), "Domain unloader"), TRUE, FALSE, &error);
+	mono_thread_set_name_internal (internal, mono_string_new (mono_domain_get (), "Domain unloader"), TRUE, FALSE, &error);
```
Comment (kumpera) [inline mono/metadata/appdomain.c]: This is wrong, we're creating a string in the domain that is going to be unloaded.

File: mono/metadata/threads.c [hunk 2]
```diff
@@ -880,11 +880,92 @@ static guint32 WINAPI start_wrapper_internal(StartInfo *start_info, gsize *stack
...
+	/* Run the actual main function of the thread */
+	res = start_wrapper_internal (start_info, &res);
+
+	mono_thread_info_exit (res);
+
+	g_assert_not_reached ();
+}
+
+static gboolean
+platform_create_thread (StartInfo *start_info, gsize *const stack_size, MonoNativeThreadId *tid)
```
Comment (kumpera) [inline mono/metadata/threads.c]: This code should not be here, it belongs to utils.

**Comment by kumpera** [comment]
-1, this worsen our portability story.

File: mono/metadata/appdomain.c (hunk 1 above)
Comment (ludovic-henry) [inline mono/metadata/appdomain.c]: No, the thread is launched in the root domain, so `mono_domain_get` is going to return the root domain in this case. This function is not run in the domain that is unloaded.

**Comment by ludovic-henry** [comment]
@kumpera I updated to keep the platform specific code in `mono-threads-{posix,windows}.c`

**Comment by ludovic-henry** [comment]
@kumpera could you please re-review?

**Pull request merged**